### ISSUE TITLE
[LINUX] 40-60 FPS instead of 240 FPS

### ISSUE THREAD

**Issue by koreanfan** [ValveSoftware/Dota-2#2749]
#### Your system information

* System information from steam (`Steam` -> `Help` -> `System Information`) in a [gist](https://gist.github.com/): ryzen 5 5600, 32gb ram, rx580, ssd sata, debian stable, steam-installer 1:1.0.0.75+ds-6 amd64, mesa-vulkan-drivers 22.3.6-1+deb12u1 amd64/i386

* Have you checked for system updates?: [Yes/No] yes
* Are you using the latest stable video driver available for your system? [Yes/No] yes
* Have you verified the game files?: [Yes/No] yes

#### Please describe your issue in as much detail as possible:
Describe what you _expected_ should happen and what _did_ happen. Please link any large pastes as a [Github Gist](https://gist.github.com/).
Launch dota. Play dota with 240fps. But after starting online match i got 40-60fps instead of 240fps. Sometimes fps back to normal(240fps) if you change game resolution to different one and revert back to needed resolution. If you change rendering option in game video settings than fps also may drop below 60. If you change game video resolution to other value and revert back then fps back to normal again. Sometimes restarting the game also help.
And i got the same problem with Counter-Strike 2.
#### Steps for reproducing this issue:

1. Launch dota
2. Play online matches or hero demo mode
3. Watch at fps counter
4. If fps in normal range then try to change video renderer options or game resolution and fps drop

**Comment by betovidal** [comment]
I have very similar specs. In my case Dota is eating up all the CPU, not the GPU as I thought. Install vntop and htop to monitor these stats.

**Comment by regulator-g** [comment]
same literally half the FPS since 2023 and also have effects and shadows on mid instead of high, is this due to some of the new lighting/effects or something has tanked performance?

happens in both steam runtime and flatpak, tried x11/wayland gamemode on/off etc

gist:
https://gist.github.com/regulator-g/c08aabaf29a71abb29c5851487625dd4

mangohud summary comparison, fps% and average in bold

2024:
0.1% Min FPS,1% Min FPS,97% Percentile FPS,Average FPS,GPU Load,CPU Load,Average Frame Time,Average GPU Temp,Average CPU Temp,Average VRAM Used,Average RAM Used,Average Swap Used,Peak GPU Load,Peak CPU Load,Peak GPU Temp,Peak CPU Temp,Peak VRAM Used,Peak RAM Used,Peak Swap Used
3.2,11.4,141.1**,59.9,57.**7,16.3,16.7,36.0,50.0,3.3,9.9,1.3,73,32.3,39,53,3.7,11.5,1.5

2023:
0.1% Min FPS,1% Min FPS,97% Percentile FPS,Average FPS,GPU Load,CPU Load
13.5,51.4,**197.0,133.**7,73.1,14.4

**Comment by betovidal** [comment]
The recent update seems to have fixed this issue for me.

**Comment by regulator-g** [comment]
its not fixed, found a larger issue tracker on the AMD drivers gitlab

https://gitlab.freedesktop.org/drm/amd/-/issues/3528

in my case using the LTS kernel fixed the performance issue for now and I did not have resizable BAR enabled

**Comment by ghost** [comment]
Disabling CSM support in my motherboard's BIOS fixed this issue for me. I went from 30-40FPS straight back to my normal of 120. Resizeable BAR enabled, CPU is Ryzen 9 5900X and GPU is RX 6900XT. No other fixes were attempted.

**Comment by eeceebee** [comment]
> Disabling CSM support in my motherboard's BIOS fixed this issue for me. I went from 30-40FPS straight back to my normal of 120. Resizeable BAR enabled, CPU is Ryzen 9 5900X and GPU is RX 6900XT. No other fixes were attempted.

EDIT:

After further testing with MangoHUD (and a few games played), this didn't really help. I'm currently averaging around 60-70 FPS, when before it was a constant 144.

For what it's worth I'm also on Kernel Version 6.10.4-1-default

**Comment by betovidal** [comment]
Yes! By the time of my previous comment I got good FPS sometimes. It "helped" changing to a different desktop, but more recently it didn't help at all.

> Disabling CSM support in my motherboard's BIOS fixed this issue for me. I went from 30-40FPS straight back to my normal of 120. Resizeable BAR enabled, CPU is Ryzen 9 5900X and GPU is RX 6900XT. No other fixes were attempted.

This fixed it.

**Comment by ghost** [comment]
I'm glad I could help. I should mention that the idea of disabling CSM came from the link provided by @regulator-g 
> https://gitlab.freedesktop.org/drm/amd/-/issues/3528

so I would also like to thank him.

**Comment by ghost** [comment]
I've both disabled CSM and enabled ReBar and even using the 6.11.4 kernel which has fixes mentioned in this https://gitlab.freedesktop.org/drm/amd/-/issues/3528 

I still have horrible performance, averaging around 60-70fps compared to 120FPS on kernels 6.9. and bellow...

**Comment by SpiritWinD86** [comment]
for users who's playing on an old systems or systems without uefi (without re-bar option in the bios settings) it's still not fixed. this problem exist since 6.10+ kernels.
new issue is created https://gitlab.freedesktop.org/drm/amd/-/issues/3844 because previous is not fixed. please upvote it there.

**Comment by baaannnaaannn** [comment]
6.10 kernel also drops performance for me
